pkg/optimizelyjson/optimizely_json_test.go

```diff
@@ -0,0 +1,198 @@
+package optimizelyjson
+
+import (
+	"github.com/stretchr/testify/suite"
+	"testing"
+)
+
+type OptimizelyJsonTestSuite struct {
+	suite.Suite
+	jsonRepr       map[string]interface{}
+	dynamicList    []interface{}
+	optimizelyJson *OptimizelyJSON
+}
+
+func (suite *OptimizelyJsonTestSuite) SetupTest() {
+
+	suite.dynamicList = []interface{}{"1", "2", 3.01, 4.23, true}
+	suite.jsonRepr = map[string]interface{}{
+		"field1": 1,
+		"field2": 2.5,
+		"field3": "three",
+		"field4": map[string]interface{}{"inner_field1": 3, "inner_field2": suite.dynamicList},
+		"field5": true,
+		"field6": nil,
+	}
+	suite.optimizelyJson = NewOptimizelyJSON(suite.jsonRepr)
+}
+
+func (suite *OptimizelyJsonTestSuite) TestToDict() {
+
+	returnValue := suite.optimizelyJson.ToDict()
+	suite.Equal(suite.jsonRepr, returnValue)
+}
+
+func (suite *OptimizelyJsonTestSuite) TestToString() {
+
+	returnValue, err := suite.optimizelyJson.ToString()
+	suite.NoError(err)
+	expected := `{"field1":1,"field2":2.5,"field3":"three","field4":{"inner_field1":3,"inner_field2":["1","2",3.01,4.23,true]},"field5":true,"field6":null}`
+	suite.Equal(expected, returnValue)
+}
+
+func (suite *OptimizelyJsonTestSuite) TestGetValueInvalidJsonKeyEmptySchema() {
+	emptyStruct := struct{}{}
+	err := suite.optimizelyJson.GetValue("some_key", &emptyStruct)
+	suite.Error(err)
+	suite.Equal(`json key "some_key" not found`, err.Error())
+	suite.Equal(struct{}{}, emptyStruct)
+}
+
+func (suite *OptimizelyJsonTestSuite) TestGetValueInvalidJsonMultipleKeyEmptySchema() {
+	emptyStruct := struct{}{}
+	err := suite.optimizelyJson.GetValue("field3.some_key", &emptyStruct)
+	suite.Error(err)
+	suite.Equal(`json key "some_key" not found`, err.Error())
+	suite.Equal(struct{}{}, emptyStruct)
+}
+
+func (suite *OptimizelyJsonTestSuite) TestGetValueValidJsonKeyEmptySchema() {
+	emptyStruct := struct{}{}
```

Comment:
Am unable to understand, `emptyStruct` is empty but when I call `GetValue` with field4, shouldn't it set `field4` data.

Reply:
maybe that test is not needed at all, I wanted to show that there is no error even from passing empty schema.

Reply:
got it. already found a test where we are asserting partial and full schema.